tools/clang_tidy/lib/clang_tidy.dart

```diff
@@ -162,17 +164,19 @@ class ClangTidy {
     return GitRepo(options.repoPath).changedFiles;
   }
 
-  /// Given a build commands json file, and the files with local changes,
+  /// Given build commands json files, and the files with local changes,
   /// compute the lint commands to run.
   @visibleForTesting
   List<Command> getLintCommandsForChangedFiles(
-    List<dynamic> buildCommandsData,
+    List<List<Object?>> buildCommandsData,
     List<io.File> changedFiles,
   ) {
-    final List<Command> buildCommands = <Command>[
-      for (final dynamic c in buildCommandsData)
-        Command.fromMap(c as Map<String, dynamic>),
-    ];
+    final Set<UniquedFilePathBuildCommand> buildCommands = <UniquedFilePathBuildCommand>{
+      for (final List<Object?> buildCommandData in buildCommandsData)
+        for (final dynamic c in buildCommandData)
+          // Only add the file once, even if included in multiple build commands json files.
```

Comment:
The same file can appear in multiple commands in the same `compile_commands.json` file. For example, a file could be compiled once with `-DMY_DEFINE`, and another time without `-DMY_DEFINE`. So, this optimization will avoid processing a file more than once, but could cause some code to be missed, even when there is only one `compile_commands.json`.

Reply:
Good point.  I made this de-duplicaton optimization thinking of how to add more targets to the pre-push hook https://github.com/flutter/engine/blob/fffaf70467e3a6abf9207678f62b54770bc641fd/tools/githooks/lib/src/pre_push_command.dart#L44 without painfully increasing the amount of time the linter takes and blocking the push. I figured if every file is run for `host_debug`, and we get some nominal coverage for the other targets, that would be better than nothing.

For now I can work on getting it `lint-all` running for every target in the recipe on presubmit with multiple invocations, if needed.  We can revisit adding more to the pre-commit push later.